### ISSUE TITLE
API detection multi class per box

### ISSUE THREAD

**Issue by Shiro-LK** [tensorflow/models#4479]
Hello,

I see on the config files this information: 

>   // If true, boxes with the same coordinates will be merged together.
>   // This is useful when each box can have multiple labels.
>   // Note that only Sigmoid classification losses should be used.
>   optional bool merge_multiple_label_boxes = 17 [default=false];
> 
>   // If true, will use multiclass scores from object annotations as ground
>   // truth. Currently only compatible with annotated image inputs.
> optional bool use_multiclass_scores = 24 [default = false];

So, if I understand well, we can predict some class for one box, but how can we do that ? I did not find any clues on the tutorial for the inference and also for the TFrecord creation. Is it possible for now to train multi class per box ?

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by Shiro-LK** [comment]
Thank for the answer.
I updated my first post.

**Comment by Shiro-LK** [comment]
@pkulzc  thank for the answer. I have a question regarding the tfrecord, how can I create it for the multi class per box ? can I use a list of labels when creating the tfrecord or do I have to duplicate the boxes and then associated the different labels ?

**Comment by pkulzc** [comment]
I think you should duplicate boxes and associated with different labels. I suggest you to print out groundtruth class vectors (should be a K-hot vector for each real boxes) to confirm it works!

**Comment by sp-ananth** [comment]
Hello, as a follow up, is there a way to do a combination of different classification heads using the current repository? 

So instead of multiple binary (sigmoid) outputs, multiple binary OR softmax outputs. 
As a very simple example: 
head 1: PLANT, classes: shrub, bush, tree.
head 2: ANIMAL: classes: cat, dog, horse
So instead of N binary classification tasks, N multi-class tasks.

The advantage would be introducing competition between closely related classes to learn better like in the case learning to detect multiple diseases and their severity levels
